dask/array/core.py

```diff
@@ -5693,6 +5693,10 @@ def _subset_to_indexed_axes(iterable):
     inblock_idxs = np.broadcast_arrays(
         *(idx - start for idx, start in zip(dict_indexes.values(), starts))
     )
+    for i, a in enumerate(inblock_idxs):
+        if len(a) > 0:
+            dtype = np.min_scalar_type(np.max(a, axis=None))
+            inblock_idxs[i] = a.astype(dtype)
```

Comment:
let's do this before broadcasting in the comprehension above and pass `copy=False` (FWIW)

Reply:
good point, done